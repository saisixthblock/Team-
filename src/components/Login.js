import React, { Component } from 'react';
import {Button, Card, CardBody, CardHeader, Col, Form, FormGroup, Input, Label,Row} from 'reactstrap';
import 'bootstrap/dist/css/bootstrap.min.css';
import {BrowserRouter as Router, Route, Routes, Link} from 'react-router-dom';
import Register from './Register';

class Login extends Component{
    state={
        email:'',
        password:'',
        emailError:'',
        passwordError:'',
    }

    handleChange=(name,value)=>{
        let errors = this.state;
        if(name  == 'email'){   
            if(value != ''){
                let model = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
                if(!model.test(value)){
                    errors.emailError ="Invalid Email ";
                    errors.email = value;
                }
                else{
                    errors.emailError ='';
                    errors.email = value;
                }
            }
            else{
                errors.emailError ="Email required.!";
                errors.email = value;
            }
        }
        if(name == 'password'){   
            if(value != ''){
                errors.passwordError ='';
                errors.password = value;
            }
            else{
                errors.passwordError ="Password Required.!";
                errors.password = value;
            }
        }
        this.setState({
            ...errors
        })
    }
    handleBlur=(e)=>{
        let errors = this.state;
        if(e.target.name == "email"){
            if(e.target.value == ''){
                errors.emailError="Email Required";
            }
            else{
                let model = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
                if(!model.test(e.target.value)){
                    errors.emailError ="Invalid Email ";
                }
                else{
                    errors.emailError='';
                }
            }
        }
        if(e.target.name == "password"){
            if(e.target.value == ''){
                errors.passwordError="Password Required"
            }
            else{
                errors.passwordError=''
            }
        }
        this.setState({...errors})
    }

    handleSumbit=(e)=>{
        //console.log(values)
        e.preventDefault();
        let emailError,passwordError;
        if(!this.state.email){
            emailError = "Email Required.!";
        }
        if(!this.state.password){
            passwordError = "Password Required.!";
        }
        this.setState({
            emailError:emailError,
            passwordError:passwordError
        }) 
    }
    render(){
        return(
            <Row>
                <Col>
                    <Card style={{marginTop:100,marginLeft:500,marginRight:400}} >
                        <CardHeader>
                        <div className='header'><b> LOGIN</b></div><br/>
                        </CardHeader>
                        <CardBody >
                            <Form onSubmit={this.handleSumbit}>
                                <Row>
                                    <Col md={6} >
                                        <FormGroup>
                                            <Label htmlFor='email'>Email:</Label>
                                            <Input type='text' 
                                                name='email'
                                                placeholder='Email Id'
                                                value={this.state.email}
                                                onChange={(e)=>this.handleChange(e.target.name, e.target.value)}
                                                onBlur={this.handleBlur}
                                            />            
                                            {this.state.emailError?<p style={{color:'red'}}>{this.state.emailError}</p>: null}
                                        </FormGroup>
                                    </Col>
                                </Row>
                                <Row>
                                    <Col md={6} >
                                        <FormGroup>
                                            <Label htmlFor='password'>Password:</Label>
                                            <Input type='password'
                                                name='password'
                                                placeholder='Password'
                                                value={this.state.password}
                                                onChange={(e)=>this.handleChange(e.target.name, e.target.value)}
                                                onBlur={this.handleBlur}
                                            />
                                            {this.state.passwordError? <p style={{color:'red'}}>{this.state.passwordError}</p>:null}
                                        </FormGroup>
                                    </Col>
                                </Row>
                                <Row>
                                    <Col>
                                        <Button type='submit' id='test-button' outline color='success'>Login</Button> &nbsp;
                                        If Not Registered? Please <Link to='/register' component={<Register/>}>Register</Link>
                                    </Col>
                                </Row>
                            </Form>
                        </CardBody>
                    </Card>
                </Col>
            </Row>
            
        )
    }
}
export default Login