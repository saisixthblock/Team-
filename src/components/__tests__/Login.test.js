import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Login from "../Login";
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Card, Form, Input } from "reactstrap";
Enzyme.configure({adapter:new Adapter() });

describe('Create a SnapShot for the login component',()=>{
    let wrapper=shallow(<Login/>);
    let instanceFun=wrapper.instance();
    let sampleData={
        email:'abcd@gmail.com',
        password:'abcd'
    };
    it('Render the component correctly',()=>{
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
    it('Check the Card count',()=>{
        expect(wrapper.find(Card).exists()).toBe(true);
        expect(wrapper.find(Card)).toHaveLength(1);
    });
       beforeEach(()=>{
           jest.spyOn(instanceFun,'handleChange')
           instanceFun.handleChange('email','test@gmail.com');
       });
       it('call the function input value',()=>{
           expect(instanceFun.handleChange).toHaveBeenCalledWith('email','test@gmail.com')
       })
       it('set the state of input field of login component',()=>{
           expect(wrapper.state().email).toEqual('test@gmail.com');
       });
        it('submit the user data',()=>{
            wrapper.setState({
                state:{...sampleData},
            });
        });
        it("update data with button clcik",()=>{
            let submitbutton=wrapper.find('#test-button');
            submitbutton.simulate('click')
            expect(wrapper.state().email).toEqual('test@gmail.com'); 
        });
        it('Checking the form tag',()=>{
            expect(wrapper.find(Form).exists()).toBe(true); // inorder to check whether the form tag exists or not.
        })
        it('Checking the form tags',()=>{
            expect(wrapper.find(Form).length).toBe(1); // inorder to find the length of the form tag i;e; how many times the form tag used.
        })
});