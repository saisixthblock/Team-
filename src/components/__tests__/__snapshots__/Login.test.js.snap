// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create a SnapShot for the login component Render the component correctly 1`] = `
<Row
  tag="div"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
      "xxl",
    ]
  }
>
  <Col
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Card
      style={
        Object {
          "marginLeft": 500,
          "marginRight": 400,
          "marginTop": 100,
        }
      }
      tag="div"
    >
      <CardHeader
        tag="div"
      >
        <div
          className="header"
        >
          <b>
             LOGIN
          </b>
        </div>
        <br />
      </CardHeader>
      <CardBody
        tag="div"
      >
        <Form
          onSubmit={[Function]}
          tag="form"
        >
          <Row
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
                "xxl",
              ]
            }
          >
            <Col
              md={6}
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                  "xxl",
                ]
              }
            >
              <FormGroup
                tag="div"
              >
                <Label
                  htmlFor="email"
                  tag="label"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                      "xxl",
                    ]
                  }
                >
                  Email:
                </Label>
                <Input
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Email Id"
                  type="text"
                  value="test@gmail.com"
                />
              </FormGroup>
            </Col>
          </Row>
          <Row
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
                "xxl",
              ]
            }
          >
            <Col
              md={6}
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                  "xxl",
                ]
              }
            >
              <FormGroup
                tag="div"
              >
                <Label
                  htmlFor="password"
                  tag="label"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                      "xxl",
                    ]
                  }
                >
                  Password:
                </Label>
                <Input
                  name="password"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </FormGroup>
            </Col>
          </Row>
          <Row
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
                "xxl",
              ]
            }
          >
            <Col
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                  "xxl",
                ]
              }
            >
              <Button
                color="success"
                id="test-button"
                outline={true}
                tag="button"
                type="submit"
              >
                Login
              </Button>
               Â  If Not Registered? Please 
              <Link
                component={<Register />}
                to="/register"
              >
                Register
              </Link>
            </Col>
          </Row>
        </Form>
      </CardBody>
    </Card>
  </Col>
</Row>
`;
